addons/web/models/ir_http.py

```diff
@@ -150,24 +150,8 @@ def session_info(self):
                 # current_company should be default_company
                 "user_companies": {
                     'current_company': user.company_id.id,
-                    'allowed_companies': {
-                        comp.id: {
-                            'id': comp.id,
-                            'name': comp.name,
-                            'sequence': comp.sequence,
-                            'child_ids': (comp.child_ids & user_company_ids).ids,
-                            'parent_id': comp.parent_id.id,
-                        } for comp in user_company_ids
-                    },
-                    'disallowed_ancestor_companies': {
-                        comp.id: {
-                            'id': comp.id,
-                            'name': comp.name,
-                            'sequence': comp.sequence,
-                            'child_ids': (comp.child_ids & all_companies_in_hierarchy_sudo).ids,
-                            'parent_id': comp.parent_id.id,
-                        } for comp in disallowed_ancestor_companies_sudo
-                    },
+                    'allowed_companies': {comp.id: comp._get_session_info(user_company_ids) for comp in user_company_ids},
```

Comment:
Could you rename `user_company_ids` to `user_companies` because it is a recordset, not a collection of ids?